services/server/src/server/services/compiler/local/solidityCompiler.ts

```diff
@@ -174,26 +176,28 @@ export async function useCompiler(
 }
 
 export async function getAllMetadataAndSourcesFromSolcJson(
-  solcJson: JsonInput,
+  solc: ISolidityCompiler,
+  solcJsonInput: JsonInput,
   compilerVersion: string,
 ): Promise<PathBuffer[]> {
-  if (solcJson.language !== "Solidity")
+  if (solcJsonInput.language !== "Solidity")
     throw new Error(
-      "Only Solidity is supported, the json has language: " + solcJson.language,
+      "Only Solidity is supported, the json has language: " +
+        solcJsonInput.language,
     );
 
   const outputSelection = {
     "*": {
       "*": ["metadata"],
     },
   };
-  if (!solcJson.settings) {
-    solcJson.settings = {
+  if (!solcJsonInput.settings) {
+    solcJsonInput.settings = {
       outputSelection: outputSelection,
     };
   }
-  solcJson.settings.outputSelection = outputSelection;
-  const compiled = await useCompiler(compilerVersion, solcJson);
+  solcJsonInput.settings.outputSelection = outputSelection;
+  const compiled = await solc.compile(compilerVersion, solcJsonInput);
```

Comment:
Why here it's calling `solc.compile` instead of `useCompiler`?

Reply:
Good catch. Idk tbh. I think I saw there's an error there (`useCompiler` is missing the 2 new parameters) and I just went with this fix. Good catch, thanks

Reply:
Actually no, this function should work with any compiler, not only with the `useCompiler` function found in the same file i.e. the local solidityCompiler. So it makes sense to use `solc` there. 

The problem is, this function shouldn't have been in this file, but in the VerificaitonService or another common place. Because it can be used via the lambda compiler as well, and does not only work with the local compiler.

Reply:
Right, `getAllMetadataAndSourcesFromSolcJson` shouldn't be in that file. 

That function exists only because `CheckedContract` needs metadata as input, just to convert it back into json input. I think we can also leave it there for now and implement allowing passing json input to `CheckedContract` soon. Is there any other reason why `getAllMetadataAndSourcesFromSolcJson` exists?

Reply:
It's not there **just** to convert it back to json input but a CheckedContracts is based on a metadata. The metadata file nicely describes what sources a contract needs so we can "validate" if there are sources missing etc. Without that there's no way to find it out without compiling. With a given std-json this whole "one metadata-one CheckedContract" model fails because each *contract* (a file can have multiple contracts) is a contract to compile.